dev/com.ibm.ws.javamail.1.6/src/com/ibm/ws/javamail/internal/injection/MailSessionResourceFactoryBuilder.java

```diff
@@ -72,56 +69,10 @@ public ResourceFactory createResourceFactory(Map<String, Object> props) throws E
             annotationProps.put(prop.getKey(), value);
         }
 
-        String application = (String) annotationProps.remove("application");
-        String module = (String) annotationProps.remove("module");
-        String component = (String) annotationProps.remove("component");
-        String jndiName = (String) annotationProps.remove(MailSessionService.JNDI_NAME);
-        String description = (String) annotationProps.remove(MailSessionService.DESCRIPTION);
-        String storeProtocol = (String) annotationProps.remove(MailSessionService.STOREPROTOCOL);
-        String transportProtocol = (String) annotationProps.remove(MailSessionService.TRANSPORTPROTOCOL);
-        String host = (String) annotationProps.remove(MailSessionService.HOST);
-        String user = (String) annotationProps.remove(MailSessionService.USER);
-        String password = (String) annotationProps.remove(MailSessionService.PASSWORD);
-        String from = (String) annotationProps.remove(MailSessionService.FROM);
-        String storeProtocolClass = (String) annotationProps.remove(MailSessionService.STOREPROTOCOLCLASSNAME);
-        String transportProtocolClass = (String) annotationProps.remove(MailSessionService.TRANSPORTPROTOCOLCLASSNAME);
-        String property = (String) annotationProps.remove(MailSessionService.PROPERTY);
-        String mailSessionID = getMailSessionID(application, module, component, jndiName);
-
-        StringBuilder filter = new StringBuilder(FilterUtils.createPropertyFilter(ID, mailSessionID));
-        filter.insert(filter.length() - 1, '*');
-        // Fail if server.xml is already using the id
-        if (!removeExistingConfigurations(filter.toString()))
-            throw new IllegalArgumentException(mailSessionID); // internal error, shouldn't ever have been permitted in server.xml
```

Comment:
I suspect that the IllegalArgumentException check around the mailSessionID needs to stay in, so at a minimum the old way of handling application, module, component, and jndiName would need to remain.  The new loop should be able to handle all of the rest, though.

Reply:
think this should fix that.